sway/input/input-manager.c

```diff
@@ -282,7 +282,7 @@ bool sway_input_manager_has_focus(struct sway_input_manager *input,
 		swayc_t *container) {
 	struct sway_seat *seat = NULL;
 	wl_list_for_each(seat, &input->seats, link) {
-		if (seat->focus == container) {
+		if (seat->has_focus && sway_seat_get_focus(seat, &root_container) == container) {
```

Comment:
`sway_seat_get_focus` can just return null and then you don't have to check has_focus. Does has_focus need to exist at all?

Reply:
The design intent is a bit subtle because I'm trying to design for a specific future use case.

The "focus inactive" is emphasized because it is needed for rendering ("focused inactive" has different border colors) and determining where focus goes when entering a new container.

For instance, if I enter a container with some children and want to focus a leaf, focus should go to the "focus inactive" as defined as the last child container to receive focus within the container. In this case you use `sway_seat_get_focus(seat, &con)` for that specific container to get the right answer. In this case, it will not actually give you the absolute focused container.

The two use cases might actually be separate enough to warrant different functions. Then you can get rid of those checks for the main use case and have `sway_seat_get_focus(seat)` which acts on the root container and implicitly checks the `has_focus`, then another function what we have here which would be renamed `sway_seat_get_focus_inactive(seat, con)`.

> Does has_focus need to exist at all?

I put this in because lots of places in the code right now checks if there is actually a focused window, but I'm not sure if we can ever actually get into this situation. For instance, if there are no containers below the workspace level, then the workspace itself could get implicit focus.

If you don't see a problem with assuming that there is *always* a focused container, then we can take it out. Otherwise the algorithm we use here will always return the first thing in the focus stack.